### PR TITLE
feat: dockerfile templates advanced guides

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: docs/source/guides/server.rst [hunk 1]
```diff
@@ -58,10 +58,12 @@ Here’s an example of mounting Flask endpoints alongside BentoML
     def predict(input_json):
         return {'input_received': input_json, 'foo': 'bar'}
 
+    app = svc.asgi_app
```
Comment (parano) [inline docs/source/guides/server.rst:61]: is this a merge error from https://github.com/bentoml/BentoML/pull/2543/files ?

File: codecov.yml [hunk 2]
```diff
@@ -10,14 +10,14 @@ comment:
   show_carryforward_flags: true
 
 ignore:
-  - "bentoml/testing/**/*"
+  - "bentoml/testing"
+  - "bentoml/types.py"
   - "bentoml/_internal/types.py"
-  - "bentoml/_internal/external_typing/**/*"
+  - "bentoml/_internal/external_typing"
```
Comment (parano) [inline codecov.yml:16]: why make this change? was it not working before? could you link to related docs on codecov side?

Comment (aarnphm) [reply codecov.yml:16]: it actually never works before. https://docs.codecov.com/docs/ignoring-paths

Comment (parano) [reply codecov.yml:16]: `Please use this folder/**/*, which will exclude all files within the given folder.`

Comment (aarnphm) [reply codecov.yml:16]: Yet somehow :(
 
Screenshot 2022-06-10 at 08 45 13

**Pull request merged**